### ISSUE TITLE
Latest version 0.03.03 does not support my intel-rapl, it cannot find domain_name

### ISSUE THREAD

**Issue by LLyaudet** [ColinIanKing/powerstat#14]
Hello,
Thanks for your hard work :)
A little dump of shell commands is worth a thousand words ;)
```
laurent@laurent-GL73-8SD:~$ ls /sys/class/powercap/intel-rapl*
/sys/class/powercap/intel-rapl:
enabled  intel-rapl:0  power  subsystem  uevent

'/sys/class/powercap/intel-rapl:0':
constraint_0_max_power_uw    constraint_0_time_window_us  constraint_1_power_limit_uw  enabled         intel-rapl:0:1       name       uevent
constraint_0_name            constraint_1_max_power_uw    constraint_1_time_window_us  energy_uj       intel-rapl:0:2       power
constraint_0_power_limit_uw  constraint_1_name            device                       intel-rapl:0:0  max_energy_range_uj  subsystem

'/sys/class/powercap/intel-rapl:0:0':
constraint_0_max_power_uw  constraint_0_power_limit_uw  device   energy_uj            name   subsystem
constraint_0_name          constraint_0_time_window_us  enabled  max_energy_range_uj  power  uevent

'/sys/class/powercap/intel-rapl:0:1':
constraint_0_max_power_uw  constraint_0_power_limit_uw  device   energy_uj            name   subsystem
constraint_0_name          constraint_0_time_window_us  enabled  max_energy_range_uj  power  uevent

'/sys/class/powercap/intel-rapl:0:2':
constraint_0_max_power_uw  constraint_0_power_limit_uw  device   energy_uj            name   subsystem
constraint_0_name          constraint_0_time_window_us  enabled  max_energy_range_uj  power  uevent
laurent@laurent-GL73-8SD:~$ powerstat -h
powerstat, version 0.03.03
laurent@laurent-GL73-8SD:~$ sudo powerstat -R
[sudo] Mot de passe de laurent : 
Device does not have any RAPL domains, cannot power measure power usage
laurent@laurent-GL73-8SD:~$ 
```
I'm ready to help troubleshoot this bug (tomorrow, it is 5:08 now ;) ).
Just tell me what information you need, what I can try to do to help you :)

Best regards,
     Laurent Lyaudet

**Comment by ColinIanKing** [comment]
Which kernel are you using? And what is the hardware you are using?

**Comment by LLyaudet** [comment]
Hello :)
I upgraded Ubuntu from 23.04 to 23.10 since then.
As of now the kernel version is 6.5.0-13 and my laptop is a MSI GL73 8SD.
Here is the updated output of the commands:

```
laurent@laurent-GL73-8SD:~$ ls /sys/class/powercap/intel-rapl*
/sys/class/powercap/intel-rapl:
enabled  intel-rapl:0  power  subsystem  uevent

'/sys/class/powercap/intel-rapl:0':
constraint_0_max_power_uw    constraint_1_name            energy_uj            name
constraint_0_name            constraint_1_power_limit_uw  intel-rapl:0:0       power
constraint_0_power_limit_uw  constraint_1_time_window_us  intel-rapl:0:1       subsystem
constraint_0_time_window_us  device                       intel-rapl:0:2       uevent
constraint_1_max_power_uw    enabled                      max_energy_range_uj

'/sys/class/powercap/intel-rapl:0:0':
constraint_0_max_power_uw    constraint_0_time_window_us  energy_uj            power
constraint_0_name            device                       max_energy_range_uj  subsystem
constraint_0_power_limit_uw  enabled                      name                 uevent

'/sys/class/powercap/intel-rapl:0:1':
constraint_0_max_power_uw    constraint_0_time_window_us  energy_uj            power
constraint_0_name            device                       max_energy_range_uj  subsystem
constraint_0_power_limit_uw  enabled                      name                 uevent

'/sys/class/powercap/intel-rapl:0:2':
constraint_0_max_power_uw    constraint_0_time_window_us  energy_uj            power
constraint_0_name            device                       max_energy_range_uj  subsystem
constraint_0_power_limit_uw  enabled                      name                 uevent
laurent@laurent-GL73-8SD:~$ sudo powerstat -h
[sudo] Mot de passe de laurent : 
powerstat, version 0.03.03

usage: /snap/powerstat/169/usr/bin/powerstat [options] [delay [count]]
	-a enable all sampling collection options (-c, -f, -t and -H)
	-b redo a sample if a system is busy, considered less than 98% CPU idle
	-c show C-State statistics at end of the run
	-d specify delay before starting, default is 180 seconds
	-D show RAPL domain power measurements (enables -R option)
	-f show average CPU frequency
	-h show help
	-H show spread of measurements with power histogram
	-i specify CPU idle threshold, used in conjunction with -b
	-n no printing of table heading when screen scrolls
	-p redo a sample if we see process fork/exec/exit activity
	-r redo a sample if busy and we see process activity (same as -b -p)
	-R gather stats from Intel RAPL interface
	-s show process fork/exec/exit activity log
	-S calculate power from capacity drain using standard average
	-t show Thermal Zone temperatures (in degrees C)
	-z forcibly ignore zero power rate stats from the battery
	delay: delay between each sample, default is 10.0 seconds
	count: number of samples to take
laurent@laurent-GL73-8SD:~$ sudo powerstat -R
Device does not have any RAPL domains, cannot power measure power usage
laurent@laurent-GL73-8SD:~$ 
```
Best regards

**Comment by ColinIanKing** [comment]
Ah, OK, throw away the snap and use the .deb version.  I don't support the snap packaged powerstat anymore because it needs to access files that can only be accessed when installing it with devmode.

**Comment by LLyaudet** [comment]
Thanks :) I just did `snap remove powerstat` and then `sudo apt install powerstat` and it installed from the .deb instead of the snap.

Everything seems to work fine now :)
Now the ouput is:
```
laurent@laurent-GL73-8SD:~$ sudo powerstat -h
powerstat, version 0.03.03

usage: powerstat [options] [delay [count]]
	-a enable all sampling collection options (-c, -f, -t and -H)
	-b redo a sample if a system is busy, considered less than 98% CPU idle
	-c show C-State statistics at end of the run
	-d specify delay before starting, default is 180 seconds
	-D show RAPL domain power measurements (enables -R option)
	-f show average CPU frequency
	-h show help
	-H show spread of measurements with power histogram
	-i specify CPU idle threshold, used in conjunction with -b
	-n no printing of table heading when screen scrolls
	-p redo a sample if we see process fork/exec/exit activity
	-r redo a sample if busy and we see process activity (same as -b -p)
	-R gather stats from Intel RAPL interface
	-s show process fork/exec/exit activity log
	-S calculate power from capacity drain using standard average
	-t show Thermal Zone temperatures (in degrees C)
	-z forcibly ignore zero power rate stats from the battery
	delay: delay between each sample, default is 10.0 seconds
	count: number of samples to take
laurent@laurent-GL73-8SD:~$ sudo powerstat -R
Running for 60.0 seconds (60 samples at 1.0 second intervals).
Power measurements will start in 0 seconds time.

  Time    User  Nice   Sys  Idle    IO  Run Ctxt/s  IRQ/s Fork Exec Exit  Watts
19:09:13  50.1   0.0   9.5  14.0  26.4    1  30428  15362    0    0    0 103.81 
19:09:14  44.5   0.0   8.4  16.5  30.6   14  25837  13229    3    0    0  96.48 
19:09:15  59.6   0.0  10.4  12.5  17.5    6  21413  11205    1    0    0  92.21 
19:09:16  37.2   0.0   8.2  21.0  33.6   18  30448  13909    8    0    0  83.11 
19:09:17  83.1   0.0  11.1   1.8   3.9    3  17710  12485    0    0    1  96.54 
19:09:18  48.5   0.0   9.4  17.0  25.1    1  27444  12725    0    0    0  82.57 
19:09:19  71.6   0.0   9.6   7.2  11.6    6  22401  12660    0    0    0  96.45 
19:09:20  55.7   0.0   9.8  11.6  22.9    5  26545  13171    0    0    0  83.63 
19:09:21  62.8   0.0  10.7  10.6  15.9   17  27381  13268    0    0    0  87.71 
19:09:22  58.7   0.0   8.3  10.1  23.0   17  22797  11221    1    0    0  75.29 
19:09:23  54.7   0.0   9.5  13.1  22.6   16  40722  15980    0    0    1  94.05 
19:09:24  74.7   0.0  10.8   3.7  10.7    8  18827  11576    0    0    0  89.40 
19:09:25  55.1   0.0  10.1  13.8  21.1    2  25364  12855    0    0    0  84.82 
19:09:26  52.6   0.0   8.1  17.4  21.9   12  22543  12429    0    0    0  85.70 
19:09:27  66.0   0.0  10.8   8.4  14.8    9  20574  11233    0    0    3  76.21 
19:09:28  51.0   0.0   8.9  23.7  16.5    7  33493  12282    0    0   31  89.52 
19:09:29  59.3   0.0   8.3  17.5  14.9    9  21118  10681    0    0    0  91.20 
19:09:30  55.2   0.0   9.4  20.1  15.3   10  30274  13117    0    0    0  89.28 
19:09:31  44.4   0.0   7.2  27.2  21.1    2  23787  11856    0    0    0  80.32 
19:09:32  51.9   0.0   7.4  20.4  20.3   11  18713  11297    0    0    1  85.67 
19:09:33  53.0   0.0   8.0  21.6  17.4    4  41296  17010    0    0    1  98.45 
19:09:34  42.9   0.0   7.5  27.5  22.2   14  35763  13364    0    0    0  95.55 
19:09:35  52.7   0.0   7.8  20.9  18.6   13  19588  10675    0    0    0  79.47 
19:09:36  61.3   0.0   9.8  15.9  13.0    8  36366  14172    0    0    0  92.70 
19:09:37  66.3   0.0   9.3  12.6  11.9   13  16734   8994    2    0    0  82.53 
19:09:38  53.8   0.0   8.4  21.2  16.7   16  40871  15535    0    0    0  89.47 
19:09:39  80.2   0.0  11.8   4.2   3.7    9  18998  10580    0    0    0  89.97 
19:09:40  57.4   0.0   9.1  20.2  13.3   15  39207  12910    0    0    0  89.63 
19:09:41  58.0   0.0   8.4  18.1  15.4   14  23570  11691    2    0    4  89.56 
19:09:42  62.2   0.0   8.4  14.5  14.9   13  29509  12227    0    0    1  89.77 
19:09:43  54.0   0.0   8.6  20.8  16.6   14  39252  14395    0    0    1  90.08 
19:09:44  47.5   0.0   6.9  24.6  20.9   15  37909  16570    0    0    0  89.91 
19:09:45  62.0   0.0   8.3  16.8  12.9    6  27738  11980    0    0    0  86.25 
19:09:46  57.3   0.0   9.4  18.1  15.2   11  27829  12664    0    0    0  89.19 
19:09:47  59.7   0.0   7.9  18.8  13.6   16  25711  12673    0    0    0  94.52 
19:09:48  46.4   0.0   8.4  26.5  18.7    3  32055  13276    0    0   27  80.77 
19:09:49  79.5   0.0   9.2   6.9   4.4   18  22720  11557    1    0    3  87.94 
19:09:50  59.8   0.0  10.2  17.3  12.7    2  29070  12688    0    0    0  82.72 
19:09:51  58.6   0.0   7.4  20.6  13.4   14  29529  13092    0    0    0  91.99 
19:09:52  55.1   0.0   8.2  20.6  16.1   17  26508  12603    0    0    1  90.05 
-------- ----- ----- ----- ----- ----- ---- ------ ------ ---- ---- ---- ------ 
 Average  57.6   0.0   9.0  16.4  17.0 10.2 27701.0 12779.9  0.5  0.0  1.9  88.61 
 GeoMean  56.8   0.0   8.9  14.6  15.6  8.1 26855.6 12677.2  0.0  0.0  0.0  88.41 
  StdDev   9.9   0.0   1.2   6.3   6.2  5.3 6941.1 1643.0  1.4  0.0  6.3   5.99 
-------- ----- ----- ----- ----- ----- ---- ------ ------ ---- ---- ---- ------ 
 Minimum  37.2   0.0   6.9   1.8   3.7  1.0 16734.0 8994.0  0.0  0.0  0.0  75.29 
 Maximum  83.1   0.0  11.8  27.5  33.6 18.0 41296.0 17010.0  8.0  0.0 31.0 103.81 
-------- ----- ----- ----- ----- ----- ---- ------ ------ ---- ---- ---- ------ 
Summary:
CPU:  88.61 Watts on average with standard deviation 5.99  
Note: power read from RAPL domains: uncore, pkg-0, core, dram.
These readings do not cover all the hardware in this device.
^Claurent@laurent-GL73-8SD:~$ 

```

**Comment by ColinIanKing** [comment]
Glad it's working. I'll close the issue.

**Comment by LLyaudet** [comment]
Thank you :)
2 hours ago, I tested it while I was loading a dump of database for work or launching the tests of our DJango app.
And the power consumption was varying but it had spikes above 110 W.
Now I tested it while doing boinc computing.
It seems the power consumption of my graphics card is not reported and the power consumption is very stable at 86 or 87 W which must be the CPU only.
I think the spikes at 110 W were from the CPU + SSD since loading the database was write intensive.
Is there any possibility to include the power consumption of the graphics card ? If so I will open another issue for that :)
